static/js/settings_emoji.js

```diff
@@ -66,6 +67,34 @@ function sort_author_full_name(a, b) {
     return -1;
 }
 
+function add_realm_emoji(emoji_name, formData, status_widget, emoji_widget) {
+    channel.post({
+        url: "/json/realm/emoji/" + encodeURIComponent(emoji_name),
+        data: formData,
+        cache: false,
+        processData: false,
+        contentType: false,
+        success() {
+            $("#admin-emoji-status").hide();
+            ui_report.success(i18n.t("Custom emoji added!"), status_widget);
+            $("form.admin-emoji-form input[type='text']").val("");
+            $("#admin_emoji_submit").prop("disabled", false);
+            emoji_widget.clear();
+        },
+        error(xhr) {
+            $("#admin-emoji-status").hide();
+            const errors = JSON.parse(xhr.responseText).msg;
+            xhr.responseText = JSON.stringify({msg: errors});
+            ui_report.error(i18n.t("Failed"), xhr, status_widget);
+            $("#admin_emoji_submit").prop("disabled", false);
+        },
+    });
+}
```

Comment:
Can you make the extraction of this function a separate commit that has no other changes?  That's best practice when moving code, since it makes it easy for reviewers to see what you actually changed.